web/static/js/learning-maps/detail.js

```
@@ -0,0 +1,363 @@
+
+        // Progress bar
+        const progressContainer = document.createElement('div');
+        progressContainer.className = 'w-full bg-gray-200 dark:bg-gray-700 rounded-full h-2.5 mb-4';
+
+        const progressBar = document.createElement('div');
+        // Sanitize color value by using a whitelist approach
+        const safeColors = ['blue-600', 'green-600', 'yellow-600', 'purple-600', 'red-600'];
+        const safeColor = safeColors.includes(node.color) ? node.color : 'blue-600';
+        progressBar.className = `bg-${safeColor} h-2.5 rounded-full`;
+        // Ensure progress is a number between 0-100
+        const safeProgress = Math.min(Math.max(parseFloat(node.progress) || 0, 0), 100);
```

Biome (1.9.4)

[error] Use Number.parseFloat instead of the equivalent global.

ES2015 moved some globals into the Number namespace for consistency.
Safe fix: Use Number.parseFloat instead.


(lint/style/useNumberNamespace)